packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -371,21 +359,21 @@ private void launchTakeImageWithCameraIntent() {
       useFrontCamera(intent);
     }
 
-    boolean canTakePhotos = intentResolver.resolveActivity(intent);
-
-    if (!canTakePhotos) {
-      finishWithError("no_available_camera", "No cameras available for taking pictures.");
-      return;
-    }
-
     File imageFile = createTemporaryWritableImageFile();
     pendingCameraMediaUri = Uri.parse("file:" + imageFile.getAbsolutePath());
 
     Uri imageUri = fileUriResolver.resolveFileProviderUriForFile(fileProviderName, imageFile);
     intent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
     grantUriPermissions(intent, imageUri);
 
-    activity.startActivityForResult(intent, REQUEST_CODE_TAKE_IMAGE_WITH_CAMERA);
+    try {
+      activity.startActivityForResult(intent, REQUEST_CODE_TAKE_IMAGE_WITH_CAMERA);
+    } catch (ActivityNotFoundException e) {
+      // If we can't delete the file again here, there's not really anything we can do about it.
+      //noinspection ResultOfMethodCallIgnored
+      imageFile.delete();
```

Comment:
this may throw `SecurityException`, so it must be catch, so it doesn't end in a bad state. 
https://docs.oracle.com/javase/7/docs/api/java/io/File.html#delete()

Reply:
Done, I think. (Not sure what to do about that potential exception really but to ignore it.)